File: Unit3G1.go
// Author: William Provost
// Version: 1.0.0
// Date: 2025-11-10
// Fileoverview: This program reads a number of cents and converts it
// into the number of dollars and cents using integer arithmetic.

package main

import "fmt"

func main() {
    // constants
    const CENTSINDOLLAR int = 100

    // variables
    var totalCents int
    var dollars int
    var cents int

    // input
    fmt.Print("Input the cents please: ")
    fmt.Scan(&totalCents)

    // process
    dollars = totalCents / CENTSINDOLLAR
    cents = totalCents % CENTSINDOLLAR

    // output
    fmt.Printf("That is %d dollars and %d cents\n", dollars, cents)
    fmt.Println("\nDone.")
}

File: Unit3G1.ts
/**
 * @author William Provost
 * @version 1.0.0
 * @date 2025-11-10
 * @fileoverview This program reads a number of cents and converts it
 * into the number of dollars and cents using integer arithmetic.
 */

// constants
const CENTS_IN_DOLLAR: number = 100;

// variables
let totalCentsAsString: string;
let totalCents: number;
let dollars: number;
let cents: number;

// input
totalCentsAsString = prompt("Input the cents please:") || "0";
totalCents = parseInt(totalCentsAsString);

// process
dollars = Math.floor(totalCents / CENTS_IN_DOLLAR);
cents = totalCents % CENTS_IN_DOLLAR;

// output
console.log("That is " + dollars + " dollars and " + cents + " cents");
console.log("\nDone.");
